### DISCUSSION TITLE
Default filter and action functions

### DISCUSSION THREAD

**Discussion by Sceptres** [YOURLS/YOURLS#3185] [Q&A]
I am currently working on a plugin and I wanted to see the default implementation of the filter, as in what code runs if there isn't a plugin using this filter. Where can I find this information?

**Comment by dgw** [comment]
If there's no plugin hooking into the filter, nothing runs.

For more information about the context of a given filter or action, search for its name in the source code. You'll find a `yourls_apply_filter` or `yourls_do_action` function call whose surroundings will tell you precisely what happens before and after that filter/action happens.

**Comment by ozh** [comment]
What @dgw said. Basically a filter is : 

```php
$some_value = yourls_apply_filter( 'filter_name', $other_value );

function yourls_apply_filter( $filter_name, $value ) {
    if $filter_name is registered : 
        return the_function_attached_to_the_filter( $value );
    else : 
        return $value;
}
```

So, nothing hooked to a filter = no code run except the `return $value` part.
